fields.go

```diff
@@ -88,6 +90,20 @@ func Error(err error) Field {
 	return NamedError("error", err)
 }
 
+// ptrFieldsTypes lists all acceptable pointer types for creating fields.
+//
+// Caveat: we can't use ~type because the type assertion switch in zap.Any will fail with a
+// custom type.
+type ptrFieldsTypes interface {
+	*string | *int | *int32 | *int64 | *uint | *uint32 | *uint64 | *float32 | *float64 | *bool | *time.Time | *time.Duration
+}
+
+// Ptr creates a field whose value is a pointer to a type that is supported by other fields functions from this package and
+// safely and explicitly represent `nil` when appropriate.
+func Ptr[T ptrFieldsTypes](key string, value T) Field {
+	return zap.Any(key, value)
```

Comment:
This just do a switch case on the types under the hood, we can't really do much better. But at least we're providing a convenient func for this and we aren't using those pointer types too often, so we're probably fine with the type assertion cost there.

Reply:
That said if we're careful with addressing types that we've wrapped here this should be fine

Reply:
alternative API: https://github.com/sourcegraph/log/pull/15